### PR TITLE
Allow `new shared FileLogger()`

### PR DESCRIPTION
**Pull request by atilaneves**

### CODE REVIEW

File: std/logger/filelogger.d [hunk 1]
```diff
@@ -80,7 +80,10 @@ class FileLogger : Logger
                                    " created in '", d,"' could not be created."));
         }
 
-        this.file_.open(this.filename, "a");
+        static if (is(This == shared))
+            () @trusted { (cast() this.file_).open(this.filename, "a"); }();
```
Comment (thewilsonator) [inline std/logger/filelogger.d:84]: Just do this all the time? Then no need to template the constructor, add a comment about why.

Comment (atilaneves) [reply std/logger/filelogger.d]: The constructor still needs to be templated, though.

**Comment by veelo** [comment]
This PR has been merged as part of #8995 .

**Comment by thewilsonator** [comment]
Fixed by https://github.com/dlang/phobos/pull/8995

**Pull request closed without merging**